src/ref.js

```diff
@@ -0,0 +1,34 @@
+import isPlainObject from './helpers/isPlainObject';
+import wrapValidator from './helpers/wrapValidator';
+
+function isNewRef(prop) {
+  if (!isPlainObject(prop)) {
+    return false;
+  }
+  const ownProperties = Object.getOwnPropertyNames(prop);
```

Comment:
can this use the `object-keys` package, or is `current` non-enumerable?

Reply:
We could, and should definitely use Object.keys here. We already use Object.key elsewhere in this package, so using the polyfill may not be necessary.

Reply:
Ah, if we already use it directly then we can do that here too, nbd.